doc/high-level.md

```diff
@@ -0,0 +1,83 @@
+| `(COMPLEX SINGLE-FLOAT)` | `COMPLEX-SINGLE-FLOAT` |
+| `(COMPLEX DOUBLE-FLOAT)` | `COMPLEX-DOUBLE-FLOAT` |
+| `(SIGNED-BYTE 32)`       | `INT32`                |
+
+The type of the elements of the tensor can be specified with the `:type` keyword, or the constructor will attempt to find an appropriate type from the given arguments. The default element type for a tensor is `DOUBLE-FLOAT`.
+
+The order of the tensor (column-major or row-major) can be specified with the `:order` keyword. This affects the underlying storage of the tensor and will affect how it carries out operations with LAPACK. 
+
+
+## Other Library Equivalents
+
+This table was adapted largely from the [NumPy Equivalent Table](https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html#linear-algebra-equivalents).
+
+### Basic Accessors
+
+| MAGICL         | MATLAB     | NumPy                   | Description                                                   |
+|----------------|------------|-------------------------|---------------------------------------------------------------|
+| `(order a)`    | `ndims(a)` | `ndim(a)` or `a.ndim`   | Get the number of dimensions of the array.                    |
+| `(size a)`     | `numel(a)` | `size(a)` or `a.size`   | Get the number of elements of the array.                      |
+| `(shape a)`    | `size(a)`  | `shape(a)` or `a.shape` | Get the shape of the array.                                   |
+| `(tref a 1 4)` | `a(2,5)`   | `a[1 4]`                | Get the element in the second row, fifth column of the array. |
+
+### Constructors
+
+| MAGICL                                          | MATLAB             | NumPy                             | Description                                                                          |
+|-------------------------------------------------|--------------------|-----------------------------------|--------------------------------------------------------------------------------------|
+| `(from-list '(1d0 2d0 3d0 4d0 5d0 6d0) '(2 3))` | `[ 1 2 3; 4 5 6 ]` | `array([[1.,2.,3.], [4.,5.,6.]])` | Create a 2x3 matrix from given elements.                                             |
+| `(empty '(2 3 4))` or `(const 0d0 '(2 3 4))`    | `zeroes(2,3,4)`    | `zeroes((2,3,4))`                 | Create a 2x3x4 dimensional array of zeroes of double-float element type.             |
+| `(const 1d0 '(3 4))`                            | `ones(3,4)`        | `ones((3,4))`                     | Create a 3x4 dimensional array of ones of double-float element type.                 |
+| `(deye 1d0 '(3 3))`                             | `eye(3)`           | `eye(3)`                          | Create a 3x3 identity array of double-float element type.                            |
+| `(from-diag a (list (length a) (length a)))`    | `diag(a,0)`        | `diag(a,0)`                       | Create a square matrix from the diagonal entries in `a` with zeroes everywhere else. |
```

Comment:
The same idea from `deye` applies to `from-diag`: let the user specify which diagonal to fill.

Also the interface is way too verbose: the shape should be inferred by default from the length of `a`.

Reply:
#73